xbmc/windowing/osx/WinEventsOSXImpl.mm

```diff
@@ -324,19 +322,33 @@ - (NSEvent*)InputEventHandler:(NSEvent*)nsevent
 
 - (XBMC_Event)keyPressEvent:(CGEventRef*)event
 {
-  UniChar unicodeString[MAXUNICODESTRLEN] = {0};
   UniCharCount actualStringLength = 0;
+  // Get stringlength of event
+  CGEventKeyboardGetUnicodeString(*event, 0, &actualStringLength, NULL);
 
-  auto keycode = static_cast<CGKeyCode>(CGEventGetIntegerValueField(*event, kCGKeyboardEventKeycode));
-  CGEventKeyboardGetUnicodeString(*event, MAXUNICODESTRLEN,
+  // Create array with size of event string
+  UniChar unicodeString[actualStringLength];
+  memset(unicodeString, 0, sizeof(unicodeString));
```

Comment:
```suggestion
  UniChar unicodeString[actualStringLength] = {0};
```

Reply:
Can't default init variable length array. Already tried that, haha. Have to memset